core/solver/cb_gmres.cpp

```diff
@@ -240,18 +240,14 @@ void CbGmres<ValueType>::apply_dense_impl(
         auto next_krylov_basis = Vector::create_with_config_of(dense_b);
         std::shared_ptr<matrix::Dense<ValueType>> preconditioned_vector =
             Vector::create_with_config_of(dense_b);
-        auto hessenberg = Vector::create(
-            exec, dim<2>{krylov_dim + 1, krylov_dim * dense_b->get_size()[1]});
-        auto buffer = Vector::create(
-            exec, dim<2>{krylov_dim + 1, dense_b->get_size()[1]});
-        auto givens_sin =
-            Vector::create(exec, dim<2>{krylov_dim, dense_b->get_size()[1]});
-        auto givens_cos =
-            Vector::create(exec, dim<2>{krylov_dim, dense_b->get_size()[1]});
-        auto residual_norm_collection = Vector::create(
-            exec, dim<2>{krylov_dim + 1, dense_b->get_size()[1]});
-        auto residual_norm =
-            VectorNorms::create(exec, dim<2>{1, dense_b->get_size()[1]});
+        auto hessenberg =
+            Vector::create(exec, dim<2>{krylov_dim + 1, krylov_dim * num_rhs});
+        auto buffer = Vector::create(exec, dim<2>{krylov_dim + 1, num_rhs});
+        auto givens_sin = Vector::create(exec, dim<2>{krylov_dim, num_rhs});
+        auto givens_cos = Vector::create(exec, dim<2>{krylov_dim, num_rhs});
+        auto residual_norm_collection =
+            Vector::create(exec, dim<2>{krylov_dim + 1, num_rhs});
+        auto residual_norm = VectorNorms::create(exec, dim<2>{1, num_rhs});
```

Comment:
If I remember correctly, it was not added to the workspace because the at least the `krylov_bases` can't use the workspace (as the type is runtime dependent).

Reply:
The logic behind not adopting CB-GMRES to the workspace was that it will be done when CB-GMRES is also getting the simplified kernels. Currently, only kernel names have been changed (and `num_rhs` is used instead of `->get_size()[1]`).
I would like to adopt the workspace that in a separate PR (all at once including the `krylov_bases`).